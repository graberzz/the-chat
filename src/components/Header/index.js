import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import UserProfile from 'components/UserProfile';
import { vars, Container } from 'components/common';

const HeaderWrap = styled.div`
  margin-bottom: 10px;
  height: 70px;
  color: #fff;
  background-color: ${vars.mainColor};
  box-shadow: ${vars.boxShadow};
`;

const SContainer = Container.extend`
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 100%;
`;

const Logo = styled.div`
  font-size: 3rem;
`;

const Header = () => (
  <HeaderWrap>
    <SContainer>
      <Link to='/'>
        <Logo>theCHAT</Logo>
      </Link>
      <UserProfile />
    </SContainer>
  </HeaderWrap>
);

Header.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired,
  }),
};

export default Header;