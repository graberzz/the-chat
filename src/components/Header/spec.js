import React from 'react';
import { shallow } from 'enzyme';
import Header from './';

describe('<Header />', () => {
  it('matches snapshot', () => {
    expect(shallow(<Header />)).toMatchSnapshot();
  })
});