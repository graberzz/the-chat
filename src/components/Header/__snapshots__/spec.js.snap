// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<styled.div>
  <styled.div>
    <Link
      replace={false}
      to="/"
    >
      <styled.div>
        theCHAT
      </styled.div>
    </Link>
    <Connect(Component) />
  </styled.div>
</styled.div>
`;
